src/js/jsx/sections/libraries/LibraryList.jsx

```diff
@@ -82,39 +83,40 @@ define(function (require, exports, module) {
         },
 
         componentDidUpdate: function () {
-            var renameInput = this.refs.input;
-            
-            if (renameInput) {
+            if (this.refs.libraryNameInput) {
                 window.setTimeout(function () {
-                    renameInput.acquireFocus();
-                    renameInput.getDOMNode().focus();
-                }, 250);
+                    var libraryNameInput = this.refs.libraryNameInput;
+                    
+                    // Check the existence of the input again, in case it is unmounted after the timeout.
+                    if (libraryNameInput) {
+                        libraryNameInput.acquireFocus();
+                        libraryNameInput.getDOMNode().focus();
```

Comment:
Crazy idea: what if you set up an `onBlur` handler here to handle the case where someone starts a rename/create operation but then wanders off to do something else in another part of the UI?